spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java

```diff
@@ -419,6 +420,12 @@ public void testInvalidTimeTravelBasedOnBothAsOfAndTableIdentifier() {
             })
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("Cannot do time-travel based on both table identifier and AS OF");
+
+    // using branch_b1 in the table identifier and VERSION AS OF
+    Assertions.assertThatThrownBy(
+            () -> sql("SELECT * FROM %s.branch_b1 VERSION AS OF %s", tableName, snapshotId))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot do time-travel based on both table identifier and AS OF");
```

Comment:
I suspect this should also be a separate test case.

In general, I'm _very_ skeptical of PRs that add new code and assertions to existing test cases. We want test cases to be narrowly focused. When in doubt, add a new test method.

Reply:
I've applied this and moved this to https://github.com/apache/iceberg/pull/9219